SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenName.swift

```diff
@@ -22,6 +22,7 @@ fileprivate var screenNameManuallySet = false
 // Yes, I assume there are swift libraries to do this sort of thing, but nothing
 // I could find in the stdlib
 fileprivate var lock = NSLock()
+private var screenNameCallbacks: [(() -> Void)] = []
```

Comment:
Could we just pass the screen name as a parameter to the callbacks? Could get rid of `getCurrentScreenName` then

Reply:
No, we are not passing just screen name as a parameter to the callbacks, `updateCrashReportScreenName` of crashreporting library will pass as parameter of it. when ever screen name will changed in splunkRum libaray, this method  will be executed and will updated screen name  in crashreporting libaray's `custom data`. with in `updateCrashReportScreenName` method it will called getCurrentScreenName  to get current screen name.  `getCurrentScreenName` is used to communicate with crashreporting library. I had railsed PR of crashreporting  library as well.

Reply:
I agree with Siim - passing the new value as a parameter of the callback would allow us to drop `getCurrentScreenName`.

Reply:
if `sessionbasedsamplingratio` is out of range (0.0 , 1.0) we consider it as 100% and sampled all session. as we use 1.0 as default value. if you want any change in it then let us know.